include/oneapi/dpl/pstl/algorithm_impl.h

```diff
@@ -3280,72 +3280,78 @@ __parallel_set_op(__parallel_tag<_IsVector>, _ExecutionPolicy&& __exec, _RandomA
     typedef typename ::std::iterator_traits<_OutputIterator>::value_type _T;
 
     struct _SetRange
     {
-        _DifferenceType __pos, __len, __buf_pos;
+        _DifferenceType1 __pos, __len, __buf_pos;
         bool
         empty() const
         {
             return __len == 0;
         }
     };
 
-    const _DifferenceType __n1 = __last1 - __first1;
-    const _DifferenceType __n2 = __last2 - __first2;
+    const _DifferenceType1 __n1 = __last1 - __first1;
+    const _DifferenceType1 __n2 = __last2 - __first2;
 
     __par_backend::__buffer<_T> __buf(__size_func(__n1, __n2));
 
     return __internal::__except_handler([&__exec, __n1, __first1, __last1, __first2, __last2, __result, __comp,
                                          __size_func, __set_op, &__buf, __proj1, __proj2]() {
         auto __tmp_memory = __buf.get();
-        _DifferenceType __m{};
-        auto __scan = [=](_DifferenceType, _DifferenceType, const _SetRange& __s) { // Scan
+        _DifferenceType1 __m{};
+        auto __scan = [=](_DifferenceType1, _DifferenceType1, const _SetRange& __s) { // Scan
             if (!__s.empty())
                 __brick_move_destroy<__parallel_tag<_IsVector>>{}(__tmp_memory + __s.__buf_pos,
                                                                   __tmp_memory + (__s.__buf_pos + __s.__len),
                                                                   __result + __s.__pos, _IsVector{});
         };
         __par_backend::__parallel_strict_scan(
             __backend_tag{}, std::forward<_ExecutionPolicy>(__exec), __n1, _SetRange{0, 0, 0}, //-1, 0},
-            [=](_DifferenceType __i, _DifferenceType __len) {                                  // Reduce
+            [=](_DifferenceType1 __i, _DifferenceType1 __len) {                                  // Reduce
                 //[__b; __e) - a subrange of the first sequence, to reduce
                 _RandomAccessIterator1 __b = __first1 + __i;
                 _RandomAccessIterator1 __e = __first1 + (__i + __len);
 
                 //try searching for the first element which not equal to *__b
                 if (__b != __first1)
-                    __b = __internal::__pstl_upper_bound(__b, __last1, std::invoke(__proj1, *__b), __comp, __proj1);
+                    __b = __b + __internal::__pstl_upper_bound(__b, _DifferenceType1{0}, std::distance(__b, __last1),
+                                                               std::invoke(__proj1, *__b), __comp, __proj1);
 
                 //try searching for the first element which not equal to *__e
                 if (__e != __last1)
-                    __e = __internal::__pstl_upper_bound(__e, __last1, std::invoke(__proj1, *__e), __comp, __proj1);
+                    __e = __e + __internal::__pstl_upper_bound(__e, _DifferenceType1{0}, std::distance(__e, __last1),
+                                                               std::invoke(__proj1, *__e), __comp, __proj1);
```

Comment:
We know the iterators here support random access and can be subtracted, and `std::distance(__e, __last1)` is certainly a bit longer than `__last1 - __e`. Is it somehow better?

Reply:
According to https://en.cppreference.com/w/cpp/iterator/distance.html, the `std::distance` function make the same.
So I personally preferer use it.
But fixed.

Reply:
Thanks, I know it does the same. But in this case, there is no need for generality since we know all iterators are of the same kind and we use "iterator arithmetic" all over the place. And saving a few symbols in a line helps with better formatting.